sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -81,11 +82,16 @@ class CastExprMeta[INPUT <: CastBase](
             "\"-1.7976931348623158E308\" in both these cases the GPU returns Double.MaxValue " +
             "while CPU returns \"+Infinity\" and \"-Infinity\" respectively. To enable this " +
             s"operation on the GPU, set ${RapidsConf.ENABLE_CAST_STRING_TO_FLOAT} to true.")
-      case (_: StringType, _: TimestampType) if !conf.isCastStringToTimestampEnabled =>
-        willNotWorkOnGpu("the GPU only supports a subset of formats " +
-            "when casting strings to timestamps. Refer to the CAST documentation " +
-            "for more details. To enable this operation on the GPU, set" +
-            s" ${RapidsConf.ENABLE_CAST_STRING_TO_TIMESTAMP} to true.")
+      case (_: StringType, _: TimestampType) =>
+        if (!conf.isCastStringToTimestampEnabled) {
+          willNotWorkOnGpu("the GPU only supports a subset of formats " +
```

Comment:
When we add a new config that controls whether or not a type is allowed on the GPU for a certain operation, we also need to update the supported ops documentation with at least a ps note explaining the situation and pointing to the config.

Reply:
We have not been consistent about this anywhere. I understand your desire to have supported ops be complete, but we had this discussion before about what is the point of that documentation and how complete do we need to be. Have we decided that it should be exhaustive? If so then we should file a follow on issue to add in all of the notes that are missing. If not then we need to decide what exactly it is that we do want.

Reply:
I bring it up because QA has been filing bugs based on what they're seeing in the supported ops doc, so they're treating it as exhaustive.  If it's not supposed to be exhaustive then the doc probably needs to be more clear on that.  I'll file a followup issue to discuss what to do regarding these one-off configs and that doc.